model/lifecycle.go

```diff
@@ -969,3 +969,105 @@ func sortLayer(layer []task.Task, idToDisplayName map[string]string) []task.Task
 	}
 	return sortedLayer
 }
+
+// Given a patch version and a list of variant/task pairs, creates the set of new builds that
+// do not exist yet out of the set of pairs. No tasks are added for builds which already exist
+// (see AddNewTasksForPatch).
```

Comment:
I think the comment isn't quite right. 

I'd love it if there were more tests for this.

Reply:
I did not write this. This is simply AddNewBuildsForPatch with a different name so that it can be used by generate.